src/qtism/data/storage/xml/marshalling/OutcomeDeclarationMarshaller.php

```diff
@@ -96,8 +96,8 @@ protected function marshall(QtiComponent $component)
             $element->appendChild($lookupTableMarshaller->marshall($component->geTLookupTable()));
         }
 
-        if ($component->getExternalScored() !== null) {
-            static::setDOMElementAttribute($element, 'externalScored', ExternalScored::getNameByConstant($component->getExternalScored()));
+        if (Version::compare($version, '2.2.0', '>=') === true && $component->isExternallyScored()) {
+            $this->setDOMElementAttribute($element, 'externalScored', ExternalScored::getNameByConstant($component->getExternalScored()));
```

Comment:
Is a method `getExternalScored` existing in `QtiComponent`

Reply:
See above.